src/assets/wise5/directives/dynamic-prompt/DynamicPromptEvaluator.ts

```diff
@@ -0,0 +1,78 @@
+import { Observable, concatMap, map } from 'rxjs';
+import { Component } from '../../common/Component';
+import { PeerGroupStudentData } from '../../../../app/domain/peerGroupStudentData';
+import { PeerGroup } from '../../components/peerChat/PeerGroup';
+import { DynamicPromptComponent } from './dynamic-prompt.component';
+import { FeedbackRuleEvaluator } from '../../components/common/feedbackRule/FeedbackRuleEvaluator';
+import { Response } from '../../components/common/feedbackRule/Response';
+import { FeedbackRuleEvaluatorMultipleStudents } from '../../components/common/feedbackRule/FeedbackRuleEvaluatorMultipleStudents';
+import { FeedbackRuleComponent } from '../../components/feedbackRule/FeedbackRuleComponent';
+import { FeedbackRule } from '../../components/common/feedbackRule/FeedbackRule';
+
+export abstract class DynamicPromptEvaluator {
+  constructor(protected component: DynamicPromptComponent) {}
+
+  evaluate(referenceComponent: Component): void {
+    if (this.component.dynamicPrompt.isPeerGroupingTagSpecified()) {
+      this.evaluatePeerGroup(referenceComponent);
+    } else {
+      this.evaluatePersonal(referenceComponent);
+    }
+  }
+
+  protected getFeedbackRuleEvaluator(
+    referenceComponent: Component
+  ): FeedbackRuleEvaluator<Response[]> {
+    const evaluator = this.component.dynamicPrompt.isPeerGroupingTagSpecified()
+      ? new FeedbackRuleEvaluatorMultipleStudents(
+          new FeedbackRuleComponent(
+            this.component.dynamicPrompt.getRules(),
+            referenceComponent.content.maxSubmitCount,
+            false
+          ),
+          this.component.constraintService
+        )
+      : new FeedbackRuleEvaluator(
+          new FeedbackRuleComponent(
+            this.component.dynamicPrompt.getRules(),
+            referenceComponent.content.maxSubmitCount,
+            false
+          ),
+          this.component.constraintService
+        );
```

Comment:
Is this new FeedbackRuleComponent() the same in both cases? If so, can it be done once and set into a single variable.

```
new FeedbackRuleComponent(
  this.component.dynamicPrompt.getRules(),
  referenceComponent.content.maxSubmitCount,
  false
)
```